### PR TITLE
Format and lint

### PR DESCRIPTION
**Pull request by SleeplessByte**
- enable eslint for all exercises
- format all files _the same way_
- fix a stub (`simple-linked-list`)

### CODE REVIEW

**Comment by tejasbubane** [comment]
@SleeplessByte This is so good! :green_heart: 

> enable eslint for all exercises

Do we need the `.prettierrc` file? I am a little confused here because I don't see `prettier` anywhere in the codebase.

Will CI check for lint be added in separate PR?

**Comment by SleeplessByte** [comment]
I didn't yet decide if we wanted to add it by default, but first wanted to see your reaction hehehehe.

I'll add it to the maintainers workflow and I'll add the CI. We already have it for v3, so easy copy.

File: exercises/all-your-base/README.md [hunk 1]
```diff
@@ -15,21 +15,21 @@ represented as a sequence of digits, convert it to base **b**.
 In positional notation, a number in base **b** can be understood as a linear
 combination of powers of **b**.
 
-The number 42, *in base 10*, means:
+The number 42, _in base 10_, means:
 
-(4 * 10^1) + (2 * 10^0)
+(4 _ 10^1) + (2 _ 10^0)
```
Comment (SleeplessByte) [inline exercises/all-your-base/README.md:20]: I'll prettier ignore the README.mds

**Review by iHiD — APPROVED** [review]

**Comment by SleeplessByte** [comment]
@tejasbubane 

- It now checks formatting in CI
- It _forces_ the same consistent version of prettier for formatting and checking it
- It adds `npx babel-node scripts/format` to run prettier
- It adds instructions in the docs about that new action
- It adds instructions in the docs about how to set-up your editor to "automatically" do it.

In reality, code formatted between version 2.x and 2.y is going to be _very_ similar if not the same, so most of the time prettier will "just work" in editors. We just don't want to auto-install a version that has to be kept in sync with the repo unless that's necessary. We can later decide to add it as `devDependency` (but remove it in the exercises, like shelljs).

Bonus. On any _future_ pr, add a comment with:

```
/format
```

and it will auto-format the code and commit it. No need to install/run anything.

**Comment by tejasbubane** [comment]
Amazing work @SleeplessByte :+1:

**Pull request merged**